guava/src/test/java/com/fasterxml/jackson/datatype/guava/CacheDeserializationTest.java

```diff
@@ -0,0 +1,179 @@
+package com.fasterxml.jackson.datatype.guava;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Optional;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Unit tests for verifying deserialization of Guava's {@link Cache} type.
+ *
+ * @since 2.16
+ */
+public class CacheDeserializationTest extends ModuleTestBase {
+
+    /*
+    /**********************************************************
+    /* Set up
+    /**********************************************************
+     */
+
+    // [datatype-collections#96]
+    static class CacheProperties {
+        @JsonProperty
+        Cache<Long, Integer> cache;
+
+        @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
+        public CacheProperties(@JsonProperty("cache") Cache<Long, Integer> cache) {
+            this.cache = cache;
+        }
+    }
+
+    // [datatype-collections#96]
+    static class CacheDelegating {
+        Cache<Long, Integer> cache;
+
+        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
+        public CacheDelegating(Cache<Long, Integer> cache) {
+            this.cache = cache;
+        }
+
+        @JsonValue
+        Cache<Long, Integer> mapValue() {
+            return cache;
+        }
+    }
+
+    private enum MyEnum {
+        YAY,
+        BOO
+    }
+    
+    /*
+    /**********************************************************
+    /* Tests
+    /**********************************************************
+     */
+
+    private final ObjectMapper MAPPER = mapperWithModule();
+
+    public void testGuavaCacheApi() throws Exception {
+        Cache<String, String> cache = CacheBuilder.newBuilder().build();
+        // Cache does not allow null key
```

Comment:
I am ok with this, but in general it's not necessary to test dependencies of a library (they are assumed to work as specified).